apiserver/machinemanager/instance_information_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package machinemanager_test
+
+import (
+	"github.com/juju/errors"
+	jujutesting "github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/apiserver/machinemanager"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/apiserver/testing"
+	"github.com/juju/juju/cloud"
+	"github.com/juju/juju/constraints"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/environs/instances"
+	"github.com/juju/juju/provider/dummy"
+)
+
+type instanceTypesSuite struct{}
+
+var _ = gc.Suite(&instanceTypesSuite{})
+
+var over9kCPUCores uint64 = 9001
+
+func (p *instanceTypesSuite) TestInstanceTypes(c *gc.C) {
+	backend := mockBackend{
+		cloudSpec: environs.CloudSpec{},
+	}
+	authorizer := testing.FakeAuthorizer{Tag: names.NewUserTag("admin"),
+		EnvironManager: true}
+	itCons := constraints.Value{CpuCores: &over9kCPUCores}
+	failureCons := constraints.Value{}
+	env := mockEnviron{
+		results: map[constraints.Value]instances.InstanceTypesWithCostMetadata{
+			itCons: instances.InstanceTypesWithCostMetadata{
+				CostUnit:     "USD/h",
+				CostCurrency: "USD",
+				InstanceTypes: []instances.InstanceType{
+					{Name: "instancetype-1"},
+					{Name: "instancetype-2"}},
+			},
+		},
+	}
+	api := machinemanager.NewMachineManagerTestingAPI(&backend, authorizer)
+
+	cons := params.ModelInstanceTypesConstraints{
+		Constraints: []params.ModelInstanceTypesConstraint{{Value: &itCons}, {Value: &failureCons}},
+	}
+	fakeEnvironGet := func(st environs.EnvironConfigGetter,
+		newEnviron environs.NewEnvironFunc,
+	) (environs.Environ, error) {
+		return &env, nil
+	}
+	r, err := machinemanager.InstanceTypes(&api, fakeEnvironGet, cons)
```

Comment:
Should we also test the case no constraints are passed? In this case I'd expect an empty result.